### DISCUSSION TITLE
React-Native: Invariant Violation: View config getter callback for component `canvas` must be a function (received `undefined`).

### DISCUSSION THREAD

**Discussion by Ksmike** [jeffreylanters/react-unity-webgl#523] [Help Wanted!]
Hi,

Trying to get a WebGL unity project running in my React-Native applications, after following a tutorial on [Youtube](https://www.youtube.com/watch?v=-A6S-JMLRBM)

my App is a very very very simple, a single App.js file atm, and the Described code for the component

here's my code

```
import React from "react";
import { View } from "react-native";
import { Unity, useUnityContext } from "react-unity-webgl";

export default function App() {
  const files = {
    "dataUrl": "../build/webgl2.data.gz",
    "frameworkUrl": "../build/webgl2.framework.js.gz",
    "loaderUrl": "../build/webgl2.loader.js",
    "codeUrl": "../build/webgl2.wasm.gz",
  }
  const { unityProvider } = useUnityContext({
    loaderUrl: encodeURI(files?.loaderUrl),
    codeUrl: encodeURI(files?.codeUrl),
    frameworkUrl: encodeURI(files?.frameworkUrl),
    dataUrl: encodeURI(files?.dataUrl),
    webglContextAttributes: {
      preserveDrawingBuffer: true,
    },
  })

  return <View><Unity unityProvider={unityProvider} /></View>;
}
```

The error I am getting is:
![Screenshot 2023-09-06 at 13 47 08](https://github.com/jeffreylanters/react-unity-webgl/assets/7250341/56888c04-e541-438c-90cf-ec5226adac48)

Not sure why, can't find any documentation other than: https://heygabriel.medium.com/react-native-view-config-getter-callback-for-component-7449e1ac754b but the imports look correct, it made me wonder if react-unity-webgl was actually I compatible I with react native,

Any help would be appreciated!

**Comment by momenAboelmagd** [comment]
did you solve this issue ?

**Reply by momenAboelmagd** [reply]
@Ksmike

**Reply by Ksmike** [reply]
Sorry just re-read my code. No idea how I solved this issue. Apologies for being truly unhelpful!

**Reply by momenAboelmagd** [reply]
im using useUnityContext but i have the same issue @Ksmike

**Reply by gus-t27** [reply]
@Ksmike So did you manage to get this library to work with react-native or not?

**Reply by Ksmike** [reply]
> @Ksmike So did you manage to get this library to work with react-native or not?

No sorry!
